lib/network/httputils/paginator.go

```diff
@@ -0,0 +1,110 @@
+package httputils
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/node/runner/api/resource"
+)
+
+const DefaultMaxLimit uint64 = 100
+
+type Paginator struct {
+	request *http.Request
+	cursor  []byte
+	reverse bool
+	limit   uint64
+	err     error
+}
+
+func NewPaginator(r *http.Request) *Paginator {
+	p := &Paginator{
+		request: r,
+		limit:   DefaultMaxLimit,
+	}
+	p.parseRequest()
+	return p
+}
+
+func (p *Paginator) Limit() uint64 {
+	return p.limit
+}
+
+func (p *Paginator) Reverse() bool {
+	return p.reverse
+}
+
+func (p *Paginator) Cursor() []byte {
+	return p.cursor
+}
+
+func (p *Paginator) Error() error {
```

Comment:
Interesting design, using `Error()` instead of `p, err := NewPaginator()`. Is there any reason of this?

Reply:
I changed `p,err := ` :-)